exit/test_exit.c

```diff
@@ -0,0 +1,513 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    POSIX.1-2017 standard library functions tests
+ *    HEADER:
+ *    - stdlib.h
+ *    - unistd.h
+ *    TESTED:
+ *    - exit()
+ *    - _exit() (_Exit equivalent)
+ *    - atexit()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <errno.h>
+#include <unity_fixture.h>
+
+#define PATH1    "/tmp/testFile1"
+#define PATH2    "/tmp/testFile2"
+#define PATH3    "/tmp/testFile3"
+#define PATH4    "/tmp/testFile4"
+#define TEST_STR "test123"
+
+/* Set to random value */
+static int test_creatRes = 0x12345678;
+
+/* SIGCHLD signal handler */
+static void test_handler(int signum)
+{
+	/* Create file to confirm that handler has been invoked */
+	test_creatRes = creat(PATH1, DEFFILEMODE);
+}
+
+
+/* Functions to be registered by atexit */
+static void test_fun1(void)
+{
+	int fd, val = 1;
+
+	fd = open(PATH2, O_WRONLY | O_APPEND);
+	TEST_ASSERT_NOT_EQUAL_INT(-1, fd);
+	TEST_ASSERT_EQUAL_INT(sizeof(int), write(fd, &val, sizeof(int)));
+	close(fd);
+}
+
+
+static void test_fun2(void)
+{
+	int fd, val = 1234;
+
+	fd = open(PATH2, O_WRONLY | O_APPEND);
+	TEST_ASSERT_NOT_EQUAL_INT(-1, fd);
+	TEST_ASSERT_EQUAL_INT(sizeof(int), write(fd, &val, sizeof(int)));
+	close(fd);
+}
+
+
+TEST_GROUP(_exit);
```

Comment:
Function name is used as a label in macro. I would recommend to change it. It might backfire in the most unpredictable way.